Updated code for py3 + pep8 compatibility

DESm1th opened the pull request:
Note: there are a lot of places (mostly in the models) where I've told pycodestyle to ignore errors that use '!=' and '==' operators with Booleans and None. This is because sqlalchemy will construct incorrect queries if you don't use them, so I couldn't change it to satisfy pep8 standards without introducing bugs.

pep8speaks commented:
Hello @DESm1th! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`dashboard/__init__.py`](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/__init__.py):

> [Line 76:37](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/__init__.py#L76): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 135:1](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/__init__.py#L135): [F401](https://duckduckgo.com/?q=pep8%20F401) 'dashboard.views' imported but unused
> [Line 135:1](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/__init__.py#L135): [F401](https://duckduckgo.com/?q=pep8%20F401) 'dashboard.models' imported but unused



* In the file [`dashboard/forms.py`](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/forms.py):

> [Line 13:1](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/forms.py#L13): [F401](https://duckduckgo.com/?q=pep8%20F401) 'wtforms.SelectField' imported but unused
> [Line 13:1](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/forms.py#L13): [F401](https://duckduckgo.com/?q=pep8%20F401) 'wtforms.widgets' imported but unused
> [Line 14:21](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/forms.py#L14): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 21:1](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/forms.py#L21): [F401](https://duckduckgo.com/?q=pep8%20F401) '.models.Study' imported but unused
> [Line 21:1](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/forms.py#L21): [F401](https://duckduckgo.com/?q=pep8%20F401) '.models.Analysis' imported but unused

* In the file [`dashboard/models.py`](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py):

> [Line 74:48](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L74): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 315:46](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L315): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 333:21](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L333): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 338:21](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L338): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 376:25](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L376): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 380:25](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L380): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 529:45](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L529): [W503](https://duckduckgo.com/?q=pep8%20W503) line break before binary operator
> [Line 531:45](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L531): [W503](https://duckduckgo.com/?q=pep8%20W503) line break before binary operator
> [Line 538:80](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L538): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 545:80](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L545): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)
> [Line 551:80](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L551): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (93 > 79 characters)
> [Line 557:80](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L557): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (94 > 79 characters)
> [Line 665:49](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L665): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 702:48](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L702): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 706:49](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L706): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 936:53](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L936): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 950:25](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L950): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 1021:29](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L1021): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 1088:53](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L1088): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 1175:53](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L1175): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 1368:45](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L1368): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 1421:21](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L1421): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 1426:41](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L1426): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 1466:45](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L1466): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 1532:53](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L1532): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 1643:21](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L1643): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 1901:21](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L1901): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 1963:53](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/models.py#L1963): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent

* In the file [`dashboard/monitors.py`](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/monitors.py):

> [Line 71:41](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/monitors.py#L71): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 77:80](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/monitors.py#L77): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)



* In the file [`dashboard/queries.py`](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/queries.py):

> [Line 10:21](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/queries.py#L10): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 75:65](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/queries.py#L75): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 79:25](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/queries.py#L79): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 87:21](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/queries.py#L87): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 141:29](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/queries.py#L141): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 149:25](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/queries.py#L149): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent

* In the file [`dashboard/redcap.py`](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/redcap.py):

> [Line 31:41](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/redcap.py#L31): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 57:37](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/redcap.py#L57): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent



* In the file [`dashboard/utils.py`](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/utils.py):

> [Line 55:29](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/utils.py#L55): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent



* In the file [`dashboard/views.py`](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/views.py):

> [Line 14:1](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/views.py#L14): [F401](https://duckduckgo.com/?q=pep8%20F401) 'flask.abort' imported but unused
> [Line 14:1](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/views.py#L14): [F401](https://duckduckgo.com/?q=pep8%20F401) 'flask.g' imported but unused
> [Line 15:19](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/views.py#L15): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 18:25](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/views.py#L18): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 26:22](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/views.py#L26): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 29:21](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/views.py#L29): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 32:20](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/views.py#L32): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 37:25](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/views.py#L37): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 289:41](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/views.py#L289): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 582:13](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/views.py#L582): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent
> [Line 655:41](https://github.com/TIGRLab/dashboard/blob/c512f9a14c90a7925c9154a87e1ccb88e6de4aa1/dashboard/views.py#L655): [E126](https://duckduckgo.com/?q=pep8%20E126) continuation line over-indented for hanging indent

DESm1th commented:
What the hell is pep8speaks running to catch these things! I thought python3's pycodestyle would give the same output, but nope. And flake8 on the terminal gives a different list of errors that's not an exact subset of these ones either. Really annoying. I'll fix these.

josephmje commented:
One thing that we can consider is running black as a pre-commit hook. It's an opinionated reformatter that adheres to pep8.

DESm1th commented:
Does it? I ran it before complaining about pep8speaks and abandoned the idea because it completed ignored line length restrictions

josephmje commented:
I thought it took care of line lengths. But I think we have to factor in readability vs strictly adhering to pep8. If it makes sense, we can extend the line length to a bit higher than 79 characters .

jerdra commented:
I agree with Michael on readability > pep8.

Additional note:

 I've been using black for my auto-formatting and it's dealt with minor issues pretty well from my experience. But like you mentioned Dawn, it was unreliable with line length issues (sometimes would correct, sometimes would break my own changes, sometimes would do nothing)

DESm1th commented:
Ok, I'm done making style changes. The remaining pep8speaks errors are because I deleted some "# noqa" lines and replaced then with file specific ignores of the same errors and pep8speaks is still using the old un-updated .flake8 configuration. There arent any remaining errors when running flake8 on the command line with the current settings file, except for some issues with undefined models in the queries.py metric related functions. But I dont want to clean that mess up yet and I dont want to just rip it out (Mathu had it mostly working very recently so we can patch it up later until we have a better set up)

josephmje commented on dashboard/models.py:
```diff
@@ -1481,12 +1661,14 @@ class RedcapRecord(db.Model):
-    redcap_version = db.Column('redcap_version', db.String(10), default='7.4.2')
+    redcap_version = db.Column('redcap_version',
+                               db.String(10),
+                               default='7.4.2')
```
This doesn't need to be addressed now but is there a way to get the redcap version from the DET?  Just so we can stop hardcoding this value.

DESm1th replied:
I'm actually not sure! Let me play with that today. I think the hardcoding might have just been a tiny bit of a lazyness when Tom first added that feature that got brought forward with every update.

DESm1th replied:
Ah! Right. This particular hard coded instance is just so that we can safely have a 'not null' on the version column. dashboard/redcap.py does parse the DET for the version of the actual server. So the default version number should never be used, really.

josephmje replied:
Cool! Thank you for checking!
